server/master/src/main/java/org/apache/accumulo/master/upgrade/Upgrader9to10.java

```diff
@@ -500,4 +506,127 @@ public void upgradeDirColumns(ServerContext ctx, Ample.DataLevel level) {
   public static String upgradeDirColumn(String dir) {
     return new Path(dir).getName();
   }
+
+  /**
+   * Remove all file entries containing relative paths and replace them with absolute URI paths.
+   */
+  public static void upgradeRelativePaths(ServerContext ctx, Ample.DataLevel level) {
+    String tableName = level.metaTable();
+    AccumuloClient c = ctx;
+    Configuration hadoopConf = ctx.getHadoopConf();
+    String volumeProp = ctx.getConfiguration().get(Property.INSTANCE_VOLUMES_UPGRADE_RELATIVE);
+
+    // first pass check for relative paths - if any, check existence of the file path
+    // constructed from the upgrade property + relative path
+    if (!checkForRelativePaths(c, hadoopConf, tableName, volumeProp)) {
+      log.info("No relative paths found in {} during upgrade.", tableName);
+      return;
+    } else {
+      log.info("Relative Tablet File paths exist in {}, replacing with absolute using {}",
+          tableName, volumeProp);
+    }
+
+    // second pass, create atomic mutations to replace the relative path
+    replaceRelativePaths(c, hadoopConf, tableName, volumeProp);
+  }
+
+  /**
+   * Replace relative paths but only if the constructed absolute path exists on FileSystem
+   */
+  public static void replaceRelativePaths(AccumuloClient c, Configuration hadoopConf,
+      String tableName, String volumeProp) {
+    try (Scanner scanner = c.createScanner(tableName, Authorizations.EMPTY);
+        BatchWriter writer = c.createBatchWriter(tableName)) {
+      FileSystem fs = FileSystem.get(hadoopConf);
+
+      scanner.fetchColumnFamily(MetadataSchema.TabletsSection.DataFileColumnFamily.NAME);
+      for (Entry<Key,Value> entry : scanner) {
+        Key key = entry.getKey();
+        Text filePath = key.getColumnQualifier();
```

Comment:
could have `String metaEntry = key.getColumnQualifier().toString()`  once here and use metaEntry later in the code.